business/workloads.go

```diff
@@ -199,7 +208,22 @@ func (in *WorkloadService) getParsedLogs(namespace, name string, opts *core_v1.P
 	lines := strings.Split(podLog.Logs, "\n")
 	messages := make([]LogEntry, 0)
 
+	lineCount := int64(0)
+
+	var startTime *time.Time
+	if k8sOpts.SinceTime != nil {
+		startTime = &k8sOpts.SinceTime.Time
+	}
+
 	for _, line := range lines {
+		if tailLines != nil {
+			if *tailLines <= lineCount {
+				break
+			} else {
+				lineCount += 1
+			}
+		}
+
```

Comment:
The logic here is still not right.  I'm OK if we apply the TailLines after the k8s fetch but TailLines=N means return the *last* N lines in the queried Time range, not the first N lines.  As I said in the prior comment, you need to:

1) trim logs with time > endTime (i.e. endTime = sinceTime + duration)
2) apply TailLines to the remaining logs

I'd suggest setting your endTime above and as soon as hit a log entry with timestamp > endTime break out of the loop.  Then run backwards through the entries keeping at most TailLines entries.